Documentation/maintenance.md

```diff
@@ -0,0 +1,131 @@
+# Maintenance
+
+## Overview
+
+An etcd cluster needs periodic maintenance to remain reliable.
+Depending on an etcd application's needs, this maintenance can usually
+be automated and performed without down-time or significantly degraded performance.
+
+All etcd maintenance manages storage resources consumed by the etcd keyspace.
+Failure to adequately control the keyspace size is guarded by storage space quotas;
+if an etcd member runs low on space, a quota will trigger cluster-wide alarms
+which will put the system into a limited-operation maintenance mode.
```

Comment:
do we need to explain the `limited-opeartion` mode? maybe a link to a separation section?